File: problems/check_if_all_1's_are_at_least_length_k_places_away/solution.py
class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        count = float('inf')
        for x in nums:
            if x == 1:
                if count < k:
                    return False
                count = 0
            else: 
                count += 1
        return True

File: problems/check_if_all_1's_are_at_least_length_k_places_away/solution.go
func kLengthApart(nums []int, k int) bool {
    dist := 0
    seen := false
    for _, x := range nums {
        if x == 1 && seen && dist < k {
            return false
        }
        if x == 1 {
            dist = 0
            seen = true
        } else if x == 0 && seen {
            dist++
        }
    }
    return true
}